src/main/java/gregtech/common/command/benchmark/CommandBenchmarkLookup.java

```diff
@@ -0,0 +1,278 @@
+        }
+    }
+
+    private TrialResults trial(List<Recipe> recipeSpace, RecipeLookupFunction function,
+                               AtomicInteger failureCount) throws CommandException {
+        Recipe[] sample = new Recipe[10];
+        for (int i = 0; i < 10; i++) {
+            double r = Math.random();
+            sample[i] = recipeSpace.get((int) (r * recipeSpace.size()));
+        }
+        // no match trial
+        Set<GTRecipeInput> seen = new ObjectOpenHashSet<>();
+        List<ItemStack> items = new ObjectArrayList<>();
+        List<FluidStack> fluids = new ObjectArrayList<>();
+        for (Recipe r : sample) {
+            // if adding an item input would lead to recipe matching, do not add it.
+            if (r.getInputs().size() > 1) {
+                r.getInputs().stream().filter(i -> !seen.contains(i)).findAny()
+                        .map(GTRecipeInput::getInputStacks)
+                        .map(arr -> arr[0])
+                        .ifPresent(items::add);
+                for (Recipe value : sample) {
+                    if (value.matches(false, items, fluids)) {
+                        items.remove(items.size() - 1);
+                    }
+                }
+            }
+            // if adding a fluid input would lead to recipe matching, do not add it
+            if (r.getFluidInputs().size() > 1) {
+                r.getFluidInputs().stream().filter(i -> !seen.contains(i)).findAny()
+                        .map(GTRecipeInput::getInputFluidStack)
+                        .ifPresent(fluids::add);
+                for (Recipe recipe : sample) {
+                    if (recipe.matches(false, items, fluids)) {
+                        fluids.remove(fluids.size() - 1);
+                    }
+                }
+            }
+            // prevent randomly adding inputs in the future that would lead to matching a recipe
+            seen.addAll(r.getInputs());
+            seen.addAll(r.getFluidInputs());
+        }
+        long start = System.nanoTime();
+        Collection<Recipe> out = function.find(Long.MAX_VALUE, items, fluids);
+        // in a real situation, any outputs would be run through a count match until the first success if found.
+        // run this matching while timing to penalize returning too many possible matches
+        for (Recipe r : out) {
+            if (r.matches(false, items, fluids)) {
+                break;
+            }
+        }
+        long timeNoMatch = System.nanoTime() - start;
+        for (Recipe r : sample) {
+            if (out.contains(r)) {
+                failureCount.incrementAndGet();
+                GTLog.logger.info("[Benchmarking] Recipe sample {} has failed the no match test.", (Object) sample);
+            }
+        }
+
+        // three match excess trial
+        for (int i = 0; i < 3; i++) {
+            for (GTRecipeInput input : sample[i].getInputs()) {
+                ItemStack stack = input.getInputStacks()[0].copy();
```

Comment:
Why do we have to `copy` the Itemstack here, when above for the no match trial, in the steam, we don't copy the itemstack before adding it to the list of `items`? Same with fluids?